cmd/protoc-gen-multi/main.go

```diff
@@ -131,19 +134,44 @@ func parsePlugins(args []string) (map[string]plugin, error) {
 		value, arg, _ := strings.Cut(keyvalue, "=")
 		if strings.HasSuffix(value, "_out") {
 			name := strings.TrimSuffix(value, "_out")
-			plugin := plugins[name]
+			plugin := getPlugin(name)
 			plugin.name = name
 			plugin.out = arg
-			plugins[name] = plugin
 		} else if strings.HasSuffix(value, "_opt") {
 			name := strings.TrimSuffix(value, "_opt")
-			plugin := plugins[name]
+			plugin := getPlugin(name)
 			plugin.name = name
 			plugin.opt = arg
-			plugins[name] = plugin
 		} else {
 			return nil, fmt.Errorf("expected suffix \"_opt\" or \"_out\": %q", flag)
 		}
 	}
 	return plugins, nil
 }
+
+// generate executes the plugin with the given request and unmarshals the response.
+// Modifies both the request and response in place.
```

Comment:
We may want to surface the name of the failing plugin, maybe a deferred `retErr` check that includes the name of the plugin, e.g., `fmt.Errorf("failed to execute plugin: %s: %w, protoc-gen-name, err)`

 That way the caller has a chance to debug the specific plugin.

Reply:
Added it at the call site.